components/canvas/webgl_thread.rs

```diff
@@ -490,7 +492,25 @@ impl WebGLThread {
             .contains(ContextAttributeFlags::ALPHA);
         let texture_target = current_wr_texture_target(&self.device);
 
-        let state = Default::default();
+        let use_apple_vertex_array = WebGLImpl::needs_apple_vertex_arrays(&self.device, &ctx);
+        let default_vao = WebGLImpl::create_vertex_array(&gl, use_apple_vertex_array)
```

Comment:
This change breaks loading https://mrdoob.neocities.org/028 in ANGLE for reasons I am currently investigating. We panic with "Failed to create VAO".

Reply:
Ah - GenVertexArrays is only available in ES3 or greater, and we're using ES2 for WebGL1. I wonder if we used to take [this branch](https://github.com/servo/rust-offscreen-rendering-context/blob/1b9c74737635add2b9076e12df129cc3326c1287/src/platform/with_egl/native_gl_context.rs#L56-L67) when using ANGLE in the old implementation.

Reply:
Oh, I bet previously this didn't matter because:
* we didn't try to create a vertex array object by default
* vertex array objects are a WebGL1 extension, and ANGLE doesn't report that it supports it in ES2 contexts

We will definitely have to change one of two things here:
* always create ES3 contexts
* don't create a default VAO

Reply:
This was only necessary because of the core/compat profile thing for osmesa, right?

Reply:
Well, it's needed for any platform where we want a CORE profile, e.g. MacOS.